### ISSUE TITLE
'no-unsed-vars' with 'varsIgnorePattern' reports error in destructured object as function parameter

### ISSUE THREAD

**Issue by pavelvolek** [eslint/eslint#8749]
**Tell us about your environment**

* **ESLint Version:** 4.0.0
* **Node Version:** v7.10.0
* **npm Version:** 4.2.0

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-ESLint

Linting a code snippet:
```js
function myFunc({ age: _omit1, ...name }) {
  const { firstName: _omit2, lastName } = name;
  console.log(lastName);
}

console.log(myFunc({ firstName: "john", lastName: "doe", age: 10 }));
```

with rule:
```
'no-unused-vars': ['error', {varsIgnorePattern: '^_omit'}]
```

outputs an error:
```
error  '_omit1' is defined but never used  no-unused-vars
```

whereas the `_omit2` variable is correctly ignored.

**What did you expect to happen?**
Both variables `_omit1` and`_omit2` should be ignored by the linter.

**Comment by not-an-aardvark** [comment]
I think you're looking for the [`argsIgnorePattern`](http://eslint.org/docs/rules/no-unused-vars#argsignorepattern) option instead.

**Comment by pavelvolek** [comment]
@not-an-aardvark yes, you are correct. Thanks.
